editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -1575,6 +1574,29 @@ void VisualShaderEditor::validate_script() {
 	}
 }
 
+void VisualShaderEditor::set_current_shader_type(VisualShader::Type p_type) {
+	current_type = p_type;
+
+	String id_string = visual_shader->get_path();
+	const ResourceUID::ID uid = ResourceLoader::get_resource_uid(id_string);
```

Comment:
You can use EditorFileSystem's `get_file_uid()`.

Reply:
Just that I know, what is the benefit of that?

Reply:
`ResourceLoader` goes through all loaders until it finds a compatible and, which then extracts the UID (which may even load the file).

EditorFileSystem already has the UID indexed for each file and only needs a path lookup, so accessing it is much faster. It's only available in the editor.